src/checkers/ast.rs

```diff
@@ -1269,6 +1257,18 @@ where
                     globals,
                 })));
 
+                if self.settings.enabled.contains(&CheckCode::RET501)
+                    || self.settings.enabled.contains(&CheckCode::RET502)
+                    || self.settings.enabled.contains(&CheckCode::RET503)
+                    || self.settings.enabled.contains(&CheckCode::RET504)
+                    || self.settings.enabled.contains(&CheckCode::RET505)
+                    || self.settings.enabled.contains(&CheckCode::RET506)
+                    || self.settings.enabled.contains(&CheckCode::RET507)
+                    || self.settings.enabled.contains(&CheckCode::RET508)
+                {
+                    flake8_return::plugins::function(self, body);
+                }
```

Comment:
Could we pass in the list of globals, rather than relying on current_scope? That would remove the implicit dependence on the state of the scope stack when this is called.

Reply:
I was thinking of just passing `&globals` here, but I'm having some borrow-checker issues, because the globals are moved into the new scope which is taken by `push_scope`.

That said, I was just thinking that this example should also not trigger a `RET504`.

```python
def nonlocal_assignment():
    X = 1
    def inner():
        nonlocal X
        X = 1
        return X
```

So I'm wondering whether I even need to look into globals for this. Need to check the code.

Reply:
Have taken a different approach now.